src/Generation/enemy_type.py

```diff
@@ -0,0 +1,56 @@
+from enum import Enum
+from Generation.generator import Generator
+
+class Rarity(Enum):
+    """
+    This enum class represents the rarity levels for enemy types.
+    """
+    Common = 1,
+    Rare = 2,
+    Epic = 3,
+    Legendary = 4
+
+class EnemyType:
+    """
+    This class represents enemy types and their characteristics.
+
+    Attributes:
+        race (str): The race of the enemy type.
+        name (str): The name of the enemy type.
+        rarity (Rarity): The rarity level of the enemy type.
+    """
+    def __init__(self, race: str, name: str, rarity: Rarity):
+        self.race = race
+        self.name = name
+        self.rarity = rarity
+        #self.description: str # TODO: do we need a description?
```

Comment:
For random enemies, not necessarily. I guess we'll have to see if we're going to share any code between NPCs and randomly generated enemies.

Reply:
I think it also depends on how we want to use the unique "legendary" type, if we want to use them at all. For them it would probably be useful. I think I'll leave there for now if you don't mind and it might be deleted later.